src/components/TodoList.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useContext } from 'react';
+import classNames from 'classnames';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import { TodoItem } from './TodoItem';
+import { TodoContext } from '../contexts/TodoContext';
+import { visibleTodos } from '../utils/visibleTodos';
+
+export const TodoList: React.FC = () => {
+  const { todos, filtering, tempTodo } = useContext(TodoContext);
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      <TransitionGroup>
+        {visibleTodos(filtering, todos).map((todo) => (
+          <CSSTransition
+            key={todo.id}
+            timeout={300}
+            classNames="item"
+          >
+            <TodoItem todo={todo} key={todo.id} />
+          </CSSTransition>
+        ))}
+
+        {tempTodo && (
```

Comment:
```suggestion
        {!!tempTodo && (
```
In render with && best way use convert to boolean